### PR TITLE
Fix changing color formatting not working

### PR DESCRIPTION
**Pull request by gateguardian523**
## What
Improve GTFormattingCode

## Implementation Details
Use enumeration and fix the code that has no effect.

## Additional Information
Strangely, no one noticed that these lines of code have no effect :(

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java [hunk 1]
```diff
@@ -554,32 +554,37 @@ public class GTMachines {
     //////////////////////////////////////
     // ********* Storage *********//
     //////////////////////////////////////
+
+    public static final BiConsumer<ItemStack, List<Component>> CREATIVE_TOOLTIPS = (stack, components) -> components
+            .add(Component.translatable("gtceu.creative_tooltip.1")
+                    .append(Component.translatable("gtceu.creative_tooltip.2")
+                            .withStyle(style -> style.withColor(GTFormattingCode.RAINBOW.getCurrent())))
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java]: shouldn't this use RAINBOW_SLOW?

**Review by screret — CHANGES_REQUESTED** [review]
also, please add this to the changelog.

File: src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java (hunk 1 above)
Comment (gate-guardian) [reply src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java]: Creative Data Access Hatch uses RAINBOW, I will standardize them as RAINBOW_SLOW.

File: src/main/java/com/gregtechceu/gtceu/client/util/GTFormattingCode.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import static com.gregtechceu.gtceu.api.GTValues.ALL_COLORS;
+import static net.minecraft.ChatFormatting.AQUA;
+import static net.minecraft.ChatFormatting.DARK_GRAY;
+import static net.minecraft.ChatFormatting.GOLD;
+import static net.minecraft.ChatFormatting.GRAY;
+import static net.minecraft.ChatFormatting.RED;
+import static net.minecraft.ChatFormatting.WHITE;
+import static net.minecraft.ChatFormatting.YELLOW;
+
+public enum GTFormattingCode implements StringRepresentable {
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/client/util/GTFormattingCode.java]: actually, this shouldn't be an enum so that others can create new ones.

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/client/util/GTFormattingCode.java (hunk 2 above)
Comment (gate-guardian) [reply src/main/java/com/gregtechceu/gtceu/client/util/GTFormattingCode.java]: Okay, I will change it.

**Review by screret — APPROVED** [review]

**Pull request merged**